topics/appendix/tools/intellij_platform_gradle_plugin/tools_intellij_platform_gradle_plugin.md

```diff
@@ -0,0 +1,290 @@
+
+  intellijPlatform {
+    recommended()
+  }
+}
+
+dependencies {
+  intellijPlatform {
+    intellijIdeaCommunity("2023.3")
+  }
+}
+```
+
+The `intellijIdeaCommunity` in the previous sample is one of extension functions available for adding IntelliJ Platform dependencies to the project.
+Other IDEs can be targeted using one the extensions listed in the table below.
+
+| Extension               | Target IDE                                 |
+|-------------------------|--------------------------------------------|
+| `androidStudio`         | [Android Studio](android_studio.md)        |
+| `clion`                 | [CLion](clion.md)                          |
+| `fleetBackend`          | Fleet Backend                              |
+| `gateway`               | Gateway                                    |
+| `goland`                | [GoLand](goland.md)                        |
+| `intellijIdeaCommunity` | [IntelliJ IDEA Community Edition](idea.md) |
+| `intellijIdeaUltimate`  | [](idea_ultimate.md)                       |
+| `phpstorm`              | [PhpStorm](phpstorm.md)                    |
+| `pycharmCommunity`      | [PyCharm Community Edition](pycharm.md)    |
+| `pycharmProfessional`   | [PyCharm Professional](pycharm.md)         |
+| `rider`                 | [Rider](rider.md)                          |
+| `rustRover`             | Rust Rover                                 |
+| `writerside`            | Writerside                                 |
+
+#### Parametrized IntelliJ Platform dependency
+
+As a fallback, `intellijPlatform` extension can be used to allow dynamic configuration of the target platform, e.g., via <path>gradle.properties</path>:
+
+```
+platformType = IC
+platformVersion = 2023.3
+```
+
+The above Gradle properties can be referenced in the <path>build.gradle.kts</path> file with:
+
+```kotlin
+dependencies {
+  intellijPlatform {
+    val type = providers.gradleProperty("platformType")
+    val version = providers.gradleProperty("platformVersion")
+    
+    intellijPlatform(type, version)
+  }
+}
+```
+
+The `intellijPlatform` helper accepts also the `IntelliJPlatformType` enum type:
+
+```kotlin
+import org.jetbrains.intellij.platform.gradle.utils.IntelliJPlatformType
+
+dependencies {
+  intellijPlatform {
+    val version = providers.gradleProperty("platformVersion")
+    
+    intellijPlatform(IntelliJPlatformType.IntellijIdeaUltimate, version)
+  }
+}
+```
+
+#### Local IntelliJ Platform IDE instance
+
+It is possible to refer to the locally available IntelliJ-based IDE using the `local` helper function:
+
+```kotlin
+repositories {
+  intellijPlatform {
+    ...
+    ivy()
+  }
+}
+
+dependencies {
+  intellijPlatform {
+    local("/Users/hsz/Applications/IntelliJ IDEA Ultimate.app")
+  }
+}
+```
+
+Note that the `ivy` entry needs to be added to the `repositories {}` block as all local dependencies (local IDE, bundled plugins, etc.) rely on the Ivy repositories.
+
+### Setting up plugin dependencies
```

Comment:
Setting up Plugin Dependencies (not sure about the "up". Looks weird when capitalized)